go-controller/pkg/cni/cni.go

```diff
@@ -131,8 +135,39 @@ func (pr *PodRequest) cmdAdd(kubeAuth *KubeAPIAuth, clientset *ClientSet) (*Resp
 	}
 	// Get the IP address and MAC address of the pod
 	// for DPU, ensure connection-details is present
+
+	var (
+		hasUserDefinedPrimaryNetwork        bool
+		userDefinedPrimaryNetworkAnnotation *util.PodAnnotation
+		userDefinedNetworkName              string
+	)
 	pod, annotations, podNADAnnotation, err := GetPodWithAnnotations(pr.ctx, clientset, namespace, podName,
-		pr.nadName, annotCondFn)
+		pr.nadName, func(annotations map[string]string, nadName string) (*util.PodAnnotation, bool) {
+			// TODO: we need to simplify this function.
+			annotation, isReady := annotCondFn(annotations, nadName)
+			if nadName == "default" {
```

Comment:
this thing below should only happen **IIF** the feature flag allows it.

Since this runs on ovnkube, I think we should be able to have access to that.